tasks/unit-tests/git_api_tests.py

```diff
@@ -0,0 +1,275 @@
+import unittest
+from itertools import cycle
+from unittest import mock
+
+from invoke.exceptions import Exit
+
+from .. import release
+from ..libs.common.github_api import GithubAPI
+from ..libs.common.github_workflows import GithubException, GithubWorkflows
+from ..libs.common.gitlab import Gitlab, get_gitlab_token
+from ..libs.version import Version
+
+##################### MOCKED GITLAB #####################
+
+
+def mocked_502_gitlab_requests(*_args, **_kwargs):
+    class MockResponse:
+        def __init__(self, content, status_code):
+            self.content = content
+            self.status_code = status_code
+
+        def json(self):
+            return self.content
+
+    return MockResponse(
+        "<html>\r\n<head><title>502 Bad Gateway</title></head>\r\n<body>\r\n<center><h1>502 Bad Gateway</h1></center>\r\n</body>\r\n</html>\r\n",
+        502,
+    )
+
+
+def mocked_gitlab_project_request(*_args, **_kwargs):
+    class MockResponse:
+        def __init__(self, content, status_code):
+            self.content = content
+            self.status_code = status_code
+
+        def json(self):
+            return self.content
+
+    return MockResponse("name", 200)
+
+
```

Comment:
Could you add one more test (for each Github and Gitlab) to test when we're receiving an APIError but <500 ? This is the only case we're missing !